### PR TITLE
Issue53

### PR DESCRIPTION
**Pull request by ahmedqarmout2**
image
image
image
image
image

### CODE REVIEW

File: public/js/admin.js [hunk 1]
```diff
@@ -30,20 +30,190 @@ var displayAccountsTable = function() {
...
+
+/* display Export Accounts Form */
+var displayExportAccountsForm = function() {
+    $.ajax({
+        type: 'GET',
+        url: '/accountsExportForm',
+        success: function(data) {
+            $('#admin-content').html(data);
+
+            $('#account-export-back-button').click(function(){
```
Comment (Mohammed-JS-Khan) [inline public/js/admin.js]: space

File: public/js/admin.js [hunk 2]
```diff
@@ -30,20 +30,190 @@ var displayAccountsTable = function() {
...
+
+/* display Import Accounts Form */
+var displayImportAccountsForm = function() {
+    $.ajax({
+        type: 'GET',
+        url: '/accountsImportForm',
+        success: function(data) {
+            $('#admin-content').html(data);
+
+            $('#account-import-back-button').click(function(){
```
Comment (Mohammed-JS-Khan) [inline public/js/admin.js]: space

File: public/js/admin.js [hunk 3]
```diff
@@ -30,20 +30,190 @@ var displayAccountsTable = function() {
...
+    });
+
+    $.ajax({
+        type: 'POST',
+        url: '/accountsExportFile',
+        data: {studentsList: selected},
+        success: function(data) {
+            $('#admin-content').html(data);
+
+            $('#account-export-complete-back-button').click(function(){
```
Comment (Mohammed-JS-Khan) [inline public/js/admin.js]: space

File: public/js/admin.js [hunk 4]
```diff
@@ -30,20 +30,190 @@ var displayAccountsTable = function() {
...
+        type: 'POST',
+        url: '/accountsImportFile',
+        processData: false,
+        contentType: false,
+        data: formData,
+        success: function(data) {
+            successSnackbar('File uploaded successfully');
+            $('#admin-content').html(data);
+
+            $('#account-import-list-back-button').click(function(){
```
Comment (Mohammed-JS-Khan) [inline public/js/admin.js]: space

File: public/js/admin.js [hunk 5]
```diff
@@ -30,20 +30,190 @@ var displayAccountsTable = function() {
...
+
+    $.ajax({
+        type: 'POST',
+        url: '/accountsImportList',
+        data: {selectedList: selected},
+        success: function(data) {
+            successSnackbar('Students\' list uploaded successfully');
+            $('#admin-content').html(data);
+
+            $('#account-import-complete-back-button').click(function(){
```
Comment (Mohammed-JS-Khan) [inline public/js/admin.js]: space

File: server.js [hunk 6]
```diff
@@ -875,6 +863,204 @@ app.get('/studentsListofIdsNames', function(req, res){
     });
 });
 
+/* Display accounts export form */
+app.get('/accountsExportForm', function(req, res){
```
Comment (Mohammed-JS-Khan) [inline server.js]: space

File: server.js [hunk 7]
```diff
@@ -875,6 +863,204 @@ app.get('/studentsListofIdsNames', function(req, res){
...
+
+        return res.status(200).render('users/accounts-export-form', {
+            user: req.session.user,
+            students: studentsList
+        });
+    });
+});
+
+/* Display accounts import form */
+app.get('/accountsImportForm', function(req, res){
```
Comment (Mohammed-JS-Khan) [inline server.js]: space

File: server.js [hunk 8]
```diff
@@ -875,6 +863,204 @@ app.get('/studentsListofIdsNames', function(req, res){
...
+        return res.status(403).send('Permission Denied');
+    }
+
+    return res.status(200).render('users/accounts-import-form', {
+        user: req.session.user
+    });
+});
+
+/* Display accounts export form */
+app.post('/accountsExportFile', function(req, res){
```
Comment (Mohammed-JS-Khan) [inline server.js]: space

File: views/users/accounts-import-complete.pug [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        a.btn-floating.halfway-fab.waves-effect.waves-light.red.left#account-import-list-back-button
+          i.material-icons arrow_back
+      .card-content
+        include ../checkmark
+        h5.card-title Import Job Completed
+        h6 The list of students' accounts has been imported
+      .card-action
+        .flow-left
+          h5.alignTextLeft &nbsp;Report:
+          h6.alignTextLeft &nbsp;&nbsp;&nbsp;&nbsp;Total number of students added: #{added}
```
Comment (ahmedqarmout2) [inline views/users/accounts-import-complete.pug:14]: anyone knows a better way to do this?
I want to do
Reports
(space)stuff
(space)stuff

File: server.js [hunk 10]
```diff
@@ -875,6 +863,204 @@ app.get('/studentsListofIdsNames', function(req, res){
...
+    if (!req.session.user) {
+        return res.redirect('/');
+    }
+
+    if (req.session.user.type !== common.userTypes.ADMIN) {
+        return res.status(403).send('Permission Denied');
+    }
+
+    if (!req.body.selectedList) {
+        return res.status(400).send('Invalid students\' list');succAddition
```
Comment (Mohammed-JS-Khan) [inline server.js]: what is succAddition?

File: server.js [hunk 11]
```diff
@@ -875,6 +863,204 @@ app.get('/studentsListofIdsNames', function(req, res){
...
+                    exist: exist,
+                    total: total
+                });
+            }
+        });
+    }
+});
+
+/* download */
+app.get('/download', function(req, res){
```
Comment (Mohammed-JS-Khan) [inline server.js]: space

**Comment by Mohammed-JS-Khan** [comment]
Maybe we should have a "select all" and "deselect all" button pair? Just incase the prof has a change of heart. not really merge-blocking tho. great work fam :1st_place_medal: 

Ahmed: Nice idea. Lets ask larry in the meeting

**Review by Mohammed-JS-Khan — APPROVED** [review]

File: server.js (hunk 10 above)
Comment (ahmedqarmout2) [reply server.js]: mistake, will remove it

File: views/users/accounts-import-complete.pug (hunk 9 above)
Comment (esteeroramy) [reply views/users/accounts-import-complete.pug:14]: thought we were gunna make a space generator in `common.js`

**Review by mehtadarshan1 — APPROVED** [review]
good work

File: views/users/accounts-import-complete.pug (hunk 9 above)
Comment (Mohammed-JS-Khan) [reply views/users/accounts-import-complete.pug:14]: Thats how I would do it.

**Review by esteeroramy — APPROVED** [review]

**Pull request merged**